### ISSUE TITLE
Expand the JsonDataEditor

### ISSUE THREAD

**Issue by se5a** [Pulsar4xDevs/Pulsar4x#71]
Intercross Created an Editor to help with the Tech files, 
this would be good if it were expanded to cover other json data files, and have it show links between requirements etc.

**Comment by intercross21** [comment]
kk

**Comment by se5a** [comment]
I've added a few smaller generic controls which will be the same across the different datafiles.
They're not fully working yet, most of them get the info they require already, but none of them export it or however that's handled. 
They're all in the installation window, though the arrangement needs some love.

**Comment by se5a** [comment]
ok so the installation window seems to display all the info it needs to, load installation, mineral and tech files, open the installation window and double click an installation in the left panel. 
but:  
1: If you open the installations window, *then* load all the datafiles nothing shows. you have to load the datafiles before opening the window. don't know why this is...  
2: it will crash if you don't load all the files. (looking for guid that are not loaded)  
3: nothing in the way of saving data yet. 
4: This is an odd one which has me stumped. the TechRequrementsUC works fine in the installation panel, but if you try add this control to another the winforms designer throws up some errors about serialisation. I don't know whats up with this since it's working fine in the installation panel.

**Comment by se5a** [comment]
5: Abilities and mineral cost UCs are editable, but the edit doesnt stick.

**Comment by se5a** [comment]
I'm a little unsure about the file path in this.

each SD type has a DataHandlerAndEvents of it's own. ie:
```
public static DataHolderAndEvents<TechSD> TechData = new DataHolderAndEvents<TechSD>("Techs");
public static DataHolderAndEvents<InstallationSD> InstallationData = new DataHolderAndEvents<InstallationSD>("Installations");
public static DataHolderAndEvents<MineralSD> MineralData = new DataHolderAndEvents<MineralSD>("Minerals"); 
```
each of these have a 
```
private List<string> _loadedFiles = new List<string>();
```  
However, the way that strings are added to teh above list, it's adding it to *each* of the DataHolderAndEvents.
so:
```
Data.InstallationData.GetLoadedFiles()
```
returns a list with InstallationData.Json, TechData.Json, and Minerals.Json.

**Comment by se5a** [comment]
hey look over there! some progress!

component window is just about done. 

6: need to check saving new.
7: need to check updating old.
8: need to check clearing data and creating from scratch.
9: need to add mineral costs to component window, and more importantly the component ability SD. 
it'll need to be a list of dictionaries. or a dictionary of lists... that's going to be a pain to display in the propertygrid. 
10: need to look at the Data class. I think we might/should be able to use the ecslib datastore to do more of the work for us here. like, loading all the files.

**Comment by se5a** [comment]
Right, so I've gutted the Data class here compleatly, it now works off the ecslib StaticDataManager, which will ensure that stuff is saved as per how the StaticDataManager wants it. 
I've lost some functionality, there's some events that are no longer there... and the components window is the only one showing anything now. it also exports and imports per directory. the directory form in winforms is horrible though. 

but it builds. and runs. if someone else wants to help out getting this whole thing back up and working here...

**Comment by se5a** [comment]
Wonder if we should put this on hold till we've nailed down the StaticData a bit more.
I've just changed the TechSD, as well as compleatly re-written the componentSD stuff. 
so we've gone backwards with this.

**Comment by Auranis** [comment]
Gearing up to start refactoring the old JsonDataEditor over to the new, MVVM-format JsonDataEditor2.

**Comment by se5a** [comment]
Take a look at how I've done the colony screen if you get stuck with the
binding or layout.
On 9 Oct 2015 8:32 am, "Auranis" <notifications@github.com> wrote:
>
> Gearing up to start refactoring the old JsonDataEditor over to the new,
MVVM-format JsonDataEditor2.

**Comment by se5a** [comment]
Ouch, this one is way out of date, and no longer exists in the current UI. 
needs to be re-written for the current ui. 
Unless we keep the old one as a completely separate tool, using the old ui.
